src/widgets/immunizations/immunization-mapper.ts

```diff
@@ -0,0 +1,111 @@
+  const lotNumber = immunizationResource?.resource?.lotNumber;
+  const protocolApplied =
+    immunizationResource?.resource?.protocolApplied?.length > 0 &&
+    immunizationResource?.resource?.protocolApplied[0];
+  const sequenceLabel = protocolApplied?.series;
+  const sequenceNumber = protocolApplied?.doseNumberPositiveInt;
+  const occurrenceDateTime = dayjs(
+    immunizationResource?.resource?.occurrenceDateTime
+  ).format("YYYY-MM-DD");
+  const expirationDate = dayjs(
+    immunizationResource?.resource?.expirationDate
+  ).format("YYYY-MM-DD");
+  return {
+    encounterUuid: encounterUuid,
+    immunizationObsUuid: immunizationObsUuid,
+    manufacturer: manufacturer,
+    lotNumber: lotNumber,
+    sequenceLabel: sequenceLabel,
+    sequenceNumber: sequenceNumber,
+    occurrenceDateTime: occurrenceDateTime,
+    expirationDate: expirationDate
+  };
+};
+
+export const mapFromFhirImmunizationSearchResults = immunizationSearchResult => {
+  const groupByImmunization = groupBy(
+    immunizationSearchResult.entry,
+    "resource.vaccineCode.coding[0].code"
+  );
+  return map(groupByImmunization, (immunizationResources, key) => {
+    let doses = map(immunizationResources, mapToImmunizationDoses);
+    return {
+      vaccineName:
+        immunizationResources[0]?.resource?.vaccineCode?.coding[0].display,
+      vaccineUuid:
+        immunizationResources[0]?.resource?.vaccineCode?.coding[0].code,
+      doses: orderBy(doses, [dose => get(dose, "occurrenceDateTime")], ["desc"])
+    };
+  });
+};
+
+export const mapToFhirImmunizationResource = (
+  immunizationDose,
+  visitUuid,
+  locationUuid,
+  providerUuid
+) => {
+  const immunizationResource: FHIRImmunizationResource = {
+    resourceType: "Immunization",
+    status: "Completed",
+    id: immunizationDose.immunizationObsUuid,
+    vaccineCode: {
+      coding: [
+        {
+          system: "", //TODO: What is the proper system
+          code: immunizationDose.vaccineUuid,
+          display: immunizationDose.vaccineName
+        }
+      ]
+    },
+    patient: { id: immunizationDose.patientUuid },
+    encounter: { id: visitUuid }, //Reference of visit instead of encounter
+    occurrenceDateTime: dayjs(immunizationDose.vaccinationDate).toDate(),
+    expirationDate: dayjs(immunizationDose.expirationDate).toDate(),
+    location: { id: locationUuid },
+    performer: { actor: { id: providerUuid } },
+    manufacturer: { display: immunizationDose.manufacturer },
+    lotNumber: immunizationDose.lotNumber,
+    protocolApplied: [
+      {
+        doseNumberPositiveInt: immunizationDose.currentDose.sequenceNumber,
+        series: immunizationDose.currentDose.sequenceLabel
+      }
+    ]
+  };
+  return { resource: immunizationResource };
+};
+
+type Code = {
+  code: string;
+  system: string;
+  display: string;
+};
+
+type FHIRImmunizationResource = {
+  resourceType: "Immunization";
+  status: "Completed";
+  id: string;
+  vaccineCode: { coding: [Code] };
```

Comment:
Just to be sure - this should be a tuple with 1 entry? Or did you mean do have `Array<Code>` here?

Reply:
Generally coding is an `Array<Code>`. However the idea is to allow only one entry in the code for now, so we have fixed it to be of size one. Let me know if that makes sense.

Reply:
So if its really *always* one entry (i.e., an one-element tuple) then its fine, otherwise I would recommend to use `Array<Code>`.